### PR TITLE
added associations to models

### PR DESCRIPTION
**Pull request by Mithi-code**
### Blog App - processing data in models

- Familiarize yourself with 'rails c' command.
- Use the CRUD method to find, update and delete entities.
- Create a method called current_user in your ApplicationController.

### CODE REVIEW

File: app/models/post.rb [hunk 1]
```diff
@@ -1,3 +1,16 @@
 class Post < ApplicationRecord
```
Comment (TedLivist) [inline app/models/post.rb:1]: - [ ] There seems to be something wrong with your setup. In your schema, you have comments_conter (it would be nice to change this to `comments_counter`) and likes_counter with the `text` type. This means that our increment methods in the Like and Comment models will not work. We need to change these columns to the **`integer`** type.

  mithi-counters

  ## SUGGESTION
  You can refer to this [Stackoverflow answer](https://stackoverflow.com/a/38117274) for more elaborate details.
  Basically, you need to generate migrations, change the column and run the migration. For example:
  Run: `rails g migration change_posts_comments_conter`
  ```
  # in the db/migrate/migration_file.rb
 
  class ChangePostsCommentsConter < ActiveRecord::Migration[7.0]
    def change
      change_column :posts, :comments_conter, "integer USING comments_conter::integer"
    end
  end
  ```

- [ ] Kindly implement the same for the Post **`likes_counter`**

File: app/models/comment.rb [hunk 2]
```diff
@@ -1,4 +1,10 @@
 class Comment < ApplicationRecord
-  belongs_to :user
-  belongs_to :post
+  belongs_to :author, class_name: 'User'
+  belongs_to :post, class_name: 'Post'
+  after_save :update_comments_counter
+
+  # A method that updates the comment counter for a post.
+  def update_comments_counter
+    post.increment!(:comments_counter)
```
Comment (TedLivist) [inline app/models/comment.rb:10]: - [ ] You don't really have `comments_counter` in your Post columns. What you have is **`comments_conter`**. Kindly fix this. You can refer to this [StackOverflow answer](https://stackoverflow.com/a/1992045) on how to rename columns

File: app/models/comment.rb [hunk 3]
```diff
@@ -1,4 +1,10 @@
...
+  end
```
Comment (TedLivist) [inline app/models/comment.rb:11]: - [ ] **[OPTIONAL]** In keeping with convention, it would be nice to make this method a private method so that it is not easily accessible in any other place.

File: app/models/like.rb [hunk 4]
```diff
@@ -1,4 +1,10 @@
 class Like < ApplicationRecord
-  belongs_to :user
+  belongs_to :author, class_name: :User
   belongs_to :post
+  after_save :update_likes_counter
+
+  # A method that updates the likes counter for a post.
+  def update_likes_counter
+    posts.increment!(:likes_counter)
+  end
```
Comment (TedLivist) [inline app/models/like.rb:11]: - [ ] **[OPTIONAL]** It would be nice to make this method a private method.

File: app/models/post.rb [hunk 5]
```diff
@@ -1,3 +1,16 @@
...
-  belongs_to :user
+  belongs_to :author, class_name: 'User'
+  has_many :likes, class_name: 'like', foreign_key: 'post_id'
+  has_many :comments, class_name: 'comment', foreign_key: 'post_id'
+  after_save :update_posts_counter
+
+  # A method that updates the posts counter for a user.
+  def update_posts_counter
+    author.increment!(:posts_counter)
+  end
```
Comment (TedLivist) [inline app/models/post.rb:10]: - [ ] **[OPTIONAL]** It would be nice to make this method a private method so that it is not easily accessible any other place.

**Review by TedLivist — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ ♻️

Hi @Mithi-code,

Good job so far!
✔️ Good documentation
✔️ Implementation of ERD Diagram
However, there are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the **[OPTIONAL]** prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/models/post.rb [hunk 6]
```diff
@@ -1,3 +1,18 @@
 class Post < ApplicationRecord
```
Comment (mmsesay) [inline app/models/post.rb:1]: - [OPTIONAL] As suggested by the previous reviewer, please update the datatype for the `comments_counter` and the `likes_counter`. You might run into some errors if this is not updated going forward.

Shot |
:------:|
a1

**Review by mmsesay — APPROVED** [review]
Hi @Mithi-code,

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
